dags/hindawi/hindawi_api_client.py

```diff
@@ -2,35 +2,62 @@
 from xml.etree import ElementTree
 
 from common.request import Request
+from requests.exceptions import RequestException
+from structlog import get_logger
 
 
 class HindawiApiClient:
     def __init__(
         self,
         base_url=None,
-        path_segments=None,
+        files_url=None,
     ):
         self.base_url = base_url or getenv(
             "HINDAWI_API_BASE_URL", "https://www.hindawi.com"
         )
-        self.path_segments = path_segments or ["oai-pmh", "oai.aspx"]
+        self.files_url = files_url or getenv(
+            "HINDAWI_API_FILES_URL", "http://downloads.hindawi.com"
+        )
+        self.logger = get_logger().bind(class_name=type(self).__name__)
 
     def get_articles_metadata(self, parameters, doi=None):
+        path_segments = ["oai-pmh", "oai.aspx"]
+        empty_xml = True
         if doi:
-            self.path_segments.append(doi)
+            path_segments.append(doi)
         request = Request(
             base_url=self.base_url,
-            path_segments=self.path_segments,
+            path_segments=path_segments,
             parameters=parameters,
         )
+        if "until" not in parameters:
+            raise AttributeError("until date is required")
```

Comment:
if it's required maybe it would be better to pass it as an arg or keyword arg (not in params)?

Reply:
I have just realized that code-wise, the situation should never happen because the until date, if it's not given, has to be set to current date

Reply:
This fix has to go to another branch